ext/standard/uuid.c

```diff
@@ -0,0 +1,662 @@
+		return;
+	}
+
+	zend_update_property_ex(zend_ce_exception, &EX(This), ZSTR_KNOWN(ZEND_STR_MESSAGE), reason);
+	zend_update_property(php_ce_UUIDParseException, &EX(This), UUID_EX_INPUT_PROP, UUID_EX_INPUT_PROP_LEN, input);
+
+	if (position != NULL) {
+		zend_update_property(php_ce_UUIDParseException, &EX(This), UUID_EX_POSITION_PROP, UUID_EX_POSITON_PROP_LEN, position);
+	}
+	else {
+		zend_update_property_long(php_ce_UUIDParseException, &EX(This), UUID_EX_POSITION_PROP, UUID_EX_POSITON_PROP_LEN, 0);
+	}
+
+	if (previous != NULL) {
+		zend_update_property_ex(zend_ce_exception, &EX(This), ZSTR_KNOWN(ZEND_STR_PREVIOUS), previous);
+	}
+}
+ZEND_BEGIN_ARG_INFO_EX(UUIDParseException___construct_args, NULL, 0, 2)
+	ZEND_ARG_TYPE_INFO(0, reason, IS_STRING, 0)
+	ZEND_ARG_TYPE_INFO(0, input, IS_STRING, 0)
+	ZEND_ARG_TYPE_INFO(0, position, IS_LONG, 0)
+	ZEND_ARG_OBJ_INFO(0, previous, Throwable, 1)
+ZEND_END_ARG_INFO()
+
+PHP_METHOD(UUIDParseException, getInput)
+{
+	if (zend_parse_parameters_none_throw() == FAILURE) {
+		return;
+	}
+
+	RETURN_ZVAL(zend_read_property(
+		php_ce_UUIDParseException,
+		&EX(This),
+		UUID_EX_INPUT_PROP,
+		UUID_EX_INPUT_PROP_LEN,
+		1,
+		NULL
+	), 1, 0);
+}
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUIDParseException_getInput_args, IS_STRING, 0)
+ZEND_END_ARG_INFO()
+
+PHP_METHOD(UUIDParseException, getPosition)
+{
+	if (zend_parse_parameters_none_throw() == FAILURE) {
+		return;
+	}
+
+	RETURN_ZVAL(zend_read_property(
+		php_ce_UUIDParseException,
+		&EX(This),
+		UUID_EX_POSITION_PROP,
+		UUID_EX_POSITON_PROP_LEN,
+		1,
+		NULL
+	), 1, 0);
+}
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUIDParseException_getPosition_args, IS_LONG, 0)
+ZEND_END_ARG_INFO()
+
+static const zend_function_entry uuid_parse_exception_methods[] = {
+	PHP_ME(UUIDParseException, __construct, UUIDParseException___construct_args, ZEND_ACC_PUBLIC)
+	PHP_ME(UUIDParseException, getInput,    UUIDParseException_getInput_args,    ZEND_ACC_PUBLIC)
+	PHP_ME(UUIDParseException, getPosition, UUIDParseException_getPosition_args, ZEND_ACC_PUBLIC)
+	PHP_FE_END
+};
+
+PHP_MINIT_FUNCTION(uuid)
+{
+	zend_class_entry ce;
+
+	INIT_CLASS_ENTRY(ce, "UUID", uuid_methods);
+	php_ce_UUID = zend_register_internal_class(&ce);
+	php_ce_UUID->ce_flags |= ZEND_ACC_FINAL;
+
+	zend_declare_class_constant_long(php_ce_UUID, "VARIANT_NCS", sizeof("VARIANT_NCS") - 1, PHP_UUID_VARIANT_NCS);
+	zend_declare_class_constant_long(php_ce_UUID, "VARIANT_RFC4122", sizeof("VARIANT_RFC4122") - 1, PHP_UUID_VARIANT_RFC4122);
+	zend_declare_class_constant_long(php_ce_UUID, "VARIANT_MICROSOFT", sizeof("VARIANT_MICROSOFT") - 1, PHP_UUID_VARIANT_MICROSOFT);
+	zend_declare_class_constant_long(php_ce_UUID, "VARIANT_FUTURE_RESERVED", sizeof("VARIANT_FUTURE_RESERVED") - 1, PHP_UUID_VARIANT_FUTURE_RESERVED);
+
+	zend_declare_class_constant_long(php_ce_UUID, "VERSION_1_TIME_BASED", sizeof("VERSION_1_TIME_BASED") - 1, PHP_UUID_VERSION_1_TIME_BASED);
```

Comment:
Could you please explain why did you define these constant names? They look verbose and unnecessarily detailed. Why not rename them as follows:

```php
public const VERSION_1 = 1;
public const VERSION_2 = 2;
public const VERSION_3 = 3;
public const VERSION_4 = 4;
public const VERSION_5 = 5;
```

Reply:
I personally love that they are this verbose: saves you a lookup

Reply:
UUID “version” numbers are semantically meaningless, so you might as well use names. And as Ocramius says, doing it this way means you get useful information at-a-glance.

Reply:
What is a bit weird is that the constant names are verbose while the named constructors just use `v4()` etc.